src/python/pants/backend/openapi/lint/spectral/rules.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from dataclasses import dataclass
+
+from pants.backend.openapi.lint.spectral.skip_field import SkipSpectralField
+from pants.backend.openapi.lint.spectral.subsystem import SpectralSubsystem
+from pants.backend.openapi.target_types import (
+    OpenApiDocumentDependenciesField,
+    OpenApiDocumentField,
+    OpenApiSourceField,
+)
+from pants.core.goals.lint import LintResult, LintTargetsRequest, Partitions
+from pants.core.util_rules.external_tool import DownloadedExternalTool, ExternalToolRequest
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import CreateDigest, Digest, FileContent, MergeDigests
+from pants.engine.platform import Platform
+from pants.engine.process import FallibleProcessResult, Process
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import FieldSet, Target, TransitiveTargets, TransitiveTargetsRequest
+from pants.util.logging import LogLevel
+from pants.util.meta import classproperty
+from pants.util.strutil import pluralize
+
+
+@dataclass(frozen=True)
+class SpectralFieldSet(FieldSet):
+    required_fields = (OpenApiDocumentField,)
+
+    sources: OpenApiDocumentField
+    dependencies: OpenApiDocumentDependenciesField
+
+    @classmethod
+    def opt_out(cls, tgt: Target) -> bool:
```

Comment:
am I missing something or this class method is not being used?

Besides, since you are already defining a `FieldSet`, a more straightforward approach would be to define the field as one of the `SpectralFieldSet` attributes (like with the other fields):

```python
class SpectralFieldSet(FieldSet):
  ...
  skip: SkipSpectralField
```

And then use the value of that field to filter field sets when generating the partitions:

```python
Partitions.single_partition([fs for fs in request.field_sets if not fs.skip.value])
```

Reply:
I use `opt_out()` because it's what the linter I based this one on uses (and other linters as well). It's being used by `FieldSet` in `is_applicable()`, which is used by `partition_request_get()` in the lint goal. The docs for `opt_out()` says:
> If `True`, the target will not match with the field set, even if it has the FieldSet's `required_fields`.

Reply:
sorry, I just learnt something by reviewing this code